bundles/org.openhab.binding.meteoalerte/src/main/java/org/openhab/binding/meteoalerte/internal/MeteoAlertIconProvider.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.meteoalerte.internal;
+
+import static org.openhab.binding.meteoalerte.internal.MeteoAlerteBindingConstants.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.meteoalerte.internal.json.ResponseFieldDTO.AlertLevel;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.types.State;
+import org.openhab.core.ui.icon.IconProvider;
+import org.openhab.core.ui.icon.IconSet;
+import org.openhab.core.ui.icon.IconSet.Format;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MeteoAlertIconProvider} is the class providing binding related icons.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@Component(immediate = true, service = { IconProvider.class })
+@NonNullByDefault
+public class MeteoAlertIconProvider implements IconProvider {
+    private static final Set<String> ICONS = Set.of(WAVE.replace("-", "_"), AVALANCHE, HEAT, FREEZE.replace("-", "_"),
+            FLOOD, SNOW, STORM, RAIN.replace("-", "_"), WIND);
+    private static final IconSet ICON_SET = new IconSet(BINDING_ID, "Météo Alerte Icons",
+            "These are the icons describing weather events", Set.of(Format.SVG));
+
+    private final Logger logger = LoggerFactory.getLogger(MeteoAlertIconProvider.class);
+    private final ClassLoader classLoader;
+
+    @Activate
+    public MeteoAlertIconProvider() {
+        classLoader = Objects.requireNonNull(MeteoAlertIconProvider.class.getClassLoader());
+    }
+
+    @Override
+    public Set<IconSet> getIconSets() {
+        return Set.of(ICON_SET);
+    }
+
+    @Override
+    public Set<IconSet> getIconSets(@Nullable Locale locale) {
+        return getIconSets();
+    }
+
+    @Override
+    public @Nullable Integer hasIcon(String category, String iconSetId, Format format) {
+        return ICONS.contains(category) && format == Format.SVG ? 7 : null;
```

Comment:
Why 7 ?

Reply:
We've got to return something, and I don't want MeteoAlert binding to be in high priority. I also observed that `hasIcon` was always called with `classic` in the `category` variable, even if I name my `categories`: `meteoalert:orage`. But maybe this will change once PR https://github.com/openhab/openhab-core/pull/3548 will be merged.

Reply:
The icon server requests all icon providers and finally use the one with the highest priority returning an icon.
What you forgot here and is very important is that you have to check that iconSetId parameter equals BINDING_ID.

> I also observed that hasIcon was always called with classic in the category variable, even if I name my categories: meteoalert:orage.

Can you please try with "oh:meteoalert:orage" ?

Reply:
Note that it will then probably not work properly in Main UI (there is something to be fixed in Main UI) but it should work well in Basic UI and the Android app.

Reply:
I confirm that from Classic UI, the iconSetId is correctly filled with "meteoalerte", so there is a bug in MainUI.

Reply:
Regarding the priority, we have to let the user customize the icons and potentially replace provided icons. The custom icon provider has the priority 5. So you icon provider should have a priority lower than 5. In my opinion, it could return priority 0 like the classic icon provider. Even with your current workaround to the MainUI bug, it will not be a problem because there is no conflct between your icon names and those in classic icon set.

Reply:
Using priority greater than 0 would make sense in case you want to provide icons with names already existing in the classic icon set.

Reply:
Starting at 0 works well also.

Reply:
What class of icon are you thinking of ? Starting at 0 seems natural for "no risk"

Reply:
> What class of icon are you thinking of ?

I don't understand the question.